tests/infer/mcmc/test_nuts.py

```diff
@@ -412,3 +411,21 @@ def model(cov):
     assert_close(kernel.inverse_mass_matrix[("w",)], w_cov, atol=0.5, rtol=0.5)
     assert_close(kernel.inverse_mass_matrix[("x", "y")], xy_cov, atol=0.5, rtol=0.5)
     assert_close(kernel.inverse_mass_matrix[("z",)], z_var, atol=0.5, rtol=0.5)
+
+
+def test_arrowhead():
+    def model(data):
+        concentration = torch.tensor([1.0, 1.0, 1.0])
+        p_latent = pyro.sample('p_latent', dist.Dirichlet(concentration))
+        pyro.sample("obs", dist.Categorical(p_latent), obs=data)
+        return p_latent
+
+    true_probs = torch.tensor([0.1, 0.6, 0.3])
+    data = dist.Categorical(true_probs).sample(sample_shape=(torch.Size((2000,))))
+    nuts_kernel = NUTS(model, jit_compile=True, ignore_jit_warnings=True)
+    nuts_kernel.mass_matrix_adapter = ArrowheadMassMatrix(head_size=1)
```

Comment:
nit: could you use `head_size=2` in this test and `concentration=torch.ones(4)`?

Reply:
@fritzo A more comprehensive test is in `test_arrowhead_mass`. I will use this dirichlet categorical model as a smoke test for gradient adaptation strategy.